report.md

```diff
@@ -64,10 +64,52 @@
+#### Testing opbjectives
+- Ensure application functions correctly across all integrated systems (backend, frontend, database)
+- Validate application security features particularly token mangement and authorization (OIDC)
+- Verify the frontend provides a seamless user experience with state management and responsive design
+- Ensure CI/CD pipelin (Github Actions) operates effectiveely, catching bugs early in the build process
+
+#### Test types and strategies
+ -  Unit testing (Backend & frontend) 
+    - Each backend service (Api endpoints, business logic, token management) will be tested individually
+    - Frontend components and utility functions will be tested for correctness using unit testing frameworks like Jest and React Testing Library.
+
+ - Integration testing
+    - Ensure the backend interacts correctly with the PostgreSQL database for CRUD operations.
+    - Verify that the frontend correctly communicates with backend REST APIs
+
+
+ - End-to-end (E2E) testing
+    - Using Cypress or playwright we are able to simulate real user interactions with the app for testing
+    - We will test critical flows such as authentication, CRUD operations and state management scenarios
+
+ - Security testing
+    - The security tests validate access control through Quarkus’s token-based authentication mechanism (OIDC).
+    - Public endpoints (/all) are tested to ensure they are accessible without authentication.
+    - Protected endpoints (/admin, /user) are restricted based on roles, ensuring that:
+        - Admin users have access to both /admin and /all, but are restricted from user-specific resources.
+        - User accounts have access to /user and proper identity checks are performed (e.g., user-specific greetings).
+    - Tests ensure that unauthorized access attempts are met with appropriate error codes (401 Unauthorized, 403 Forbidden).
+
+ - Performance Testing
+    - To evalute frontend performance we will be using Lighthouse to ensure optimal load times and responsiveness
+#### Enviroment setup for testing
+- Local development
+  - Tests are run in both the backend and frontend during local development and building to ensure  the application functions correctly across the full stack.
+  - Backend tests include unit and integration tests, while frontend tests focus on component and UI logic validation.
+  - Developers can run the tests continuously as they work on the application, ensuring immediate feedback and preventing integration issues before code is pushed to the repository.
+- CI/CD (Github Actions)
+    - Each commit triggers automated test runs. Both unit and integration tests are part of the CI pipeline, ensuring no broken code is deployed.
+    - Upon successful test completion, the build will proceed, else rollbacks or alerts are triggered.
```

Comment:
In regards to the backend, it first attempts to build, then tests. We do not deploy the backend yet. We have rollbacks nor alerts.